src/app/give-help/transport-form/transport-form.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component } from '@angular/core';
+import { defaults } from '@app/shared/utils';
+import { TransportOfferDefinitionDTO } from '@app/core/api/model/transportOfferDefinitionDTO';
+import { TransportResourceService } from '../../core/api/api/transportResource.service';
+import { PREFIXES, PHONE_NUMBER_REGEX } from '@app/shared/temporary-consts';
+
+@Component({
+  selector: 'app-transport-form',
+  templateUrl: './transport-form.component.html',
+  styleUrls: ['./transport-form.component.scss'],
+})
+export class TransportFormComponent {
+  minDate: Date = new Date();
+  PREFIXES = PREFIXES;
+  PHONE_NUMBER_REGEX = PHONE_NUMBER_REGEX;
+  phonePrefix: string = '48';
+  phoneNumber: string = '';
+  data = defaults<TransportOfferDefinitionDTO>();
+
+  constructor(private transportResourceService: TransportResourceService) {}
+
+  onPhoneNumberChange(): void {
+    // Waiting for TransportOfferDefinitionDTO receive a phoneNumber prop
+    // this.data.phoneNumber = this.phonePrefix + this.phoneNumber;
+  }
+
+  submitOffer(): void {
+    this.transportResourceService.createTransport(this.data).subscribe((response) => {});
```

Comment:
jeśli jest tylko ok 200 z backendu to wystarczy subscribe();
Do tego warto by dać pipe(take(1)), żeby od razu odpieło się od nasłuchiwania na tą subskrybcje.

Reply:
There is no need to unsubscribe from HttpClient observables as they are finite observables, i.e. they complete after emitting a value.

Reply:
Do tego będzie chyba potrzebna obsługa stosownego komunikatu i ewentualny redirect na moje konto?